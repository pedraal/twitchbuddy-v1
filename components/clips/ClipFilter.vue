<template>
  <v-text-field
    v-model="value.keyword"
    :label="$t('clips.keyword')"
    outlined
    dense
    rounded
    hide-details
  />
</template>

<script>
export default {
  props: {
    value: {
      type: Object,
      default: () => {
        return {}
      },
      required: true
    }
  },
  watch: {
    value () {
      this.$emit('input', this.value)
    }
  }

}
</script>
