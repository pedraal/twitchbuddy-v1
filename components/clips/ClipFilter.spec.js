import Vuetify from 'vuetify'

import { mount, createLocalVue } from '@vue/test-utils'
import ClipFilter from './ClipFilter'

const localVue = createLocalVue()

const $t = () => {}

describe('ClipFilter', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(ClipFilter, { localVue, vuetify, mocks: { $t }, propsData: { value: { keyword: '' } } })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(ClipFilter, { localVue, vuetify, mocks: { $t }, propsData: { value: { keyword: '' } } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit an event when the field is updated', () => {
    const wrapper = mount(ClipFilter, { localVue, vuetify, mocks: { $t }, propsData: { value: { keyword: '' } } })

    const event = jest.fn()
    const field = wrapper.find('.v-text-field')

    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    wrapper.vm.$on('input', event)

    expect(event).toHaveBeenCalledTimes(0)

    // Simulate an input on the field
    field.vm.$emit('input', 'some value')

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1)
  })
})
