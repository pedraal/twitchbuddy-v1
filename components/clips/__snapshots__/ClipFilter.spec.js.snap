// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClipFilter renders correctly 1`] = `
<div class="v-input v-input--hide-details v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--rounded">
  <div class="v-input__control">
    <div class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span>â€‹</span></legend>
      </fieldset>
      <div class="v-text-field__slot"><input name="keyword" id="input-7" type="text"></div>
    </div>
  </div>
</div>
`;
